python_modules/dagster-graphql/dagster_graphql/implementation/execution/backfill.py

```diff
@@ -318,3 +324,75 @@ def resume_partition_backfill(
 
     graphene_info.context.instance.update_backfill(backfill.with_status(BulkActionStatus.REQUESTED))
     return GrapheneResumeBackfillSuccess(backfill_id=backfill_id)
+
+
+def retry_partition_backfill(
+    graphene_info: "ResolveInfo", backfill_id: str
+) -> "GrapheneRetryBackfillSuccess":
+    from ...schema.backfill import GrapheneRetryBackfillSuccess
+
+    backfill = graphene_info.context.instance.get_backfill(backfill_id)
+    if not backfill:
+        check.failed(f"No backfill found for id: {backfill_id}")
+
+    if backfill.status not in BULK_ACTION_COMPLETED_STATUSES:
+        raise DagsterInvariantViolationError(
+            f"Cannot retry backfill {backfill_id} because it is still in progress."
+        )
+
+    if backfill.is_asset_backfill:
+        if (
+            backfill.asset_backfill_data.failed_and_downstream_subset.num_partitions_and_non_partitioned_assets
+            == 0
+        ):
+            raise DagsterInvariantViolationError(
+                "Cannot retry an asset backfill that has no failed assets."
+            )
+        asset_graph = graphene_info.context.asset_graph
+        assert_permission_for_asset_graph(
+            graphene_info,
+            asset_graph,
+            backfill.asset_backfill_data.failed_and_downstream_subset.asset_keys,
+            Permissions.LAUNCH_PARTITION_BACKFILL,
+        )
+
+        new_backfill = PartitionBackfill.from_asset_graph_subset(
+            backfill_id=make_new_backfill_id(),
+            asset_graph_subset=backfill.asset_backfill_data.failed_and_downstream_subset,
+            dynamic_partitions_store=graphene_info.context.instance,
+            tags={
+                **backfill.tags,
+                PARENT_RUN_ID_TAG: backfill.backfill_id,
```

Comment:
Copying my comment from the description here 

When a retried backfill is created we add the parent backfill id and the root backfill id as tags like we do for run retries. The odd thing about this though is that all of the sub-runs will get those tags as well, which would have UI implications for the sidebar where we show the chain of retried runs.

Reply:
@prha following up on what you mentioned during the meeting today - using the same tags as run retries here feels kinda risky, mainly because of the UI component that shows the run re-execution lineage. Most straightforward thing to do you be to use backfill specific tags to track the lineage instead